Groovy/Groovy.sublime-syntax

```diff
@@ -147,49 +147,72 @@ contexts:
     - include: keyword-operator
     - include: keyword-language
   keyword-language:
-    - match: \b(try|catch|finally|throw)\b
-      scope: keyword.control.exception.groovy
-    - match: \b(return|break|continue|default|do|while|for|switch|if|else)\b
-      scope: keyword.control.groovy
+    - match: \bthrow\b
+      scope: keyword.control.flow.throw.groovy
+    - match: \btry\b
+      scope: keyword.control.exception.try.groovy
+    - match: \bfinally\b
+      scope: keyword.control.exception.finally.groovy
+    - match: \bcatch\b
+      scope: keyword.control.exception.catch.groovy
+    - match: \bdefault\b
+      scope: keyword.control.conditional.groovy
+    - match: \bfor\b
+      scope: keyword.control.loop.for.groovy
+    - match: \bwhile\b
+      scope: keyword.control.loop.while.groovy
+    - match: \bdo\b
+      scope: keyword.control.loop.do-while.groovy
+    - match: \breturn\b
+      scope: keyword.control.flow.return.groovy
+    - match: \bbreak\b
+      scope: keyword.control.flow.break.groovy
+    - match: \bcontinue\b
+      scope: keyword.control.flow.continue.groovy
+    - match: \bif\b
+      scope: keyword.control.conditional.if.groovy
+    - match: \belse\b
+      scope: keyword.control.conditional.else.groovy
+    - match: \bswitch\b
+      scope: keyword.control.conditional.switch.groovy
     - match: \bcase\b
-      scope: keyword.control.groovy
+      scope: keyword.control.conditional.case.groovy
       push:
         - meta_scope: meta.case.groovy
         - match: ":"
           scope: punctuation.definition.case-terminator.groovy
           pop: true
         - include: groovy-code-minus-map-keys
-    - match: \b(new)\b
-      scope: keyword.other.new.groovy
-    - match: \b(assert)\s
-      captures:
-        1: keyword.control.assert.groovy
+    - match: \bnew\b
+      scope: keyword.operator.new.groovy
+    - match: \bassert\b
+      scope: keyword.control.exception.assert.groovy
       push:
         - meta_scope: meta.declaration.assertion.groovy
         - match: $|(?=\})
           pop: true
         - match: ":"
           scope: keyword.operator.assert.expression-separator.groovy
         - include: groovy-code-minus-map-keys
-    - match: \b(throws)\b
-      scope: keyword.other.throws.groovy
+    - match: \bthrows\b
+      scope: keyword.control.exception.groovy
   keyword-operator:
-    - match: \b(as)\b
-      scope: keyword.operator.as.groovy
-    - match: \b(is)\b
-      scope: keyword.operator.is.groovy
+    - match: \bas\b
+      scope: keyword.context.resource.groovy
```

Comment:
I know `as` keyword was topic of discussions multiple times.

Does it make sense to interpret it as kind of `keyword.operator.assignment.[as|alias]`?

It is most often used to ...
1. import modules under a custom local name (`import foo as bar`)
2. assign variables another local name (`define foo as bar`, ...)

All of which look like defining alias names for given variables, with just the syntactical difference of assignee being located on the right.

Reply:
It is also used in Python to:
3. set the scope of a context manager, syntactic sugar for try, enter context, do work, finally exit context. 

To be fair I don't know how Groovy uses it, I'm not familiar enough with it yet, but I was thinking on type 3 in which case I think the `context.resource` subscope is more accurate than a simple alias would imply.
I certainly do like your proposal for other cases though. (Except I wonder if the assignee being on the right will be confusing if highlighted the same as other assignments as color schemes would do by default until this specific rule is added...)

Reply:
Python creates over-eager special scopes for all sorts of contexts, which doesn't need to be correct.

My thoughts apply to python and all other syntaxes as well. Context managers instantiate a class and assign it to a variable on the right, which is also some sort of alias for the context manager to refer to.

Actually I have been applying `keyword.operator.assignment` to all `as` keywords, when tweaking syntaxes as well as scoping `in` `keyword.operator.iteration`.

Reply:
Original Groovy scopes `as` `keyword.operator.as`, because it is a [coercion operator](https://groovy-lang.org/operators.html#_coercion_operator) according to Groovy's language spec. It is a kind of type cast operator.

Hence I'd hesitate to change the `keyword.operator` scope.

If a sematic scope is desired maybe 
- `keyword.operator.type.coercion` 
- or `keyword.operator.type.as`
- or `keyword.operator.assignment.type`

I've also always struggled with scoping `instanceof` keywords, which is also supported by Groovy. They could be scoped `keyword.operator.comparison.type` to re-use existing 3rd-level scope or `keyword.operator.type.instanceof`.